lnwallet/channel_test.go

```diff
@@ -7515,91 +7515,155 @@ func TestForceCloseBorkedState(t *testing.T) {
 }
 
 // TestChannelMaxFeeRate ensures we correctly compute a channel initiator's max
-// fee rate based on an allocation and its available balance. It should never
-// dip below the established fee floor.
+// fee rate based on an allocation and its available balance. When a very low
+// fee allocation value is selected the max fee rate is always floored at the
+// current fee rate of the channel.
 func TestChannelMaxFeeRate(t *testing.T) {
 	t.Parallel()
 
-	assertMaxFeeRate := func(c *LightningChannel, maxAlloc float64,
-		expFeeRate chainfee.SatPerKWeight) {
-
-		maxFeeRate := c.MaxFeeRate(maxAlloc)
-		if maxFeeRate != expFeeRate {
-			t.Fatalf("expected max fee rate of %v with max "+
-				"allocation of %v, got %v", expFeeRate,
-				maxAlloc, maxFeeRate)
-		}
-
-		if err := c.validateFeeRate(maxFeeRate); err != nil {
-			t.Fatalf("fee rate validation failed: %v", err)
-		}
-	}
-
 	// propertyTest tests that the validateFeeRate function always passes
 	// for the output returned by MaxFeeRate for any valid random inputs
 	// fed to MaxFeeRate.
 	propertyTest := func(c *LightningChannel) func(alloc maxAlloc) bool {
 		return func(ma maxAlloc) bool {
-			maxFeeRate := c.MaxFeeRate(float64(ma))
+			maxFeeRate, _ := c.MaxFeeRate(float64(ma))
 			return c.validateFeeRate(maxFeeRate) == nil
 		}
 	}
 
-	aliceChannel, _, err := CreateTestChannels(
+	// Create a non-anchor and an anchor channel setup.
+	nonAnchorChannel, _, err := CreateTestChannels(
 		t, channeldb.SingleFunderTweaklessBit,
 	)
 	require.NoError(t, err, "unable to create test channels")
 
-	if err := quick.Check(propertyTest(aliceChannel), nil); err != nil {
-		t.Fatal(err)
-	}
-
-	assertMaxFeeRate(aliceChannel, 1.0, 676794154)
-	assertMaxFeeRate(aliceChannel, 0.001, 676794)
-	assertMaxFeeRate(aliceChannel, 0.000001, 676)
-	assertMaxFeeRate(aliceChannel, 0.0000001, chainfee.FeePerKwFloor)
-
-	// Check that anchor channels are capped at their max fee rate.
 	anchorChannel, _, err := CreateTestChannels(
 		t, channeldb.SingleFunderTweaklessBit|
 			channeldb.AnchorOutputsBit|channeldb.ZeroHtlcTxFeeBit,
 	)
 	require.NoError(t, err, "unable to create test channels")
 
+	// Run the property tests for both channel types.
+	if err := quick.Check(propertyTest(nonAnchorChannel), nil); err != nil {
+		t.Fatal(err)
+	}
+
 	if err = quick.Check(propertyTest(anchorChannel), nil); err != nil {
 		t.Fatal(err)
 	}
 
-	// Anchor commitments are heavier, hence will the same allocation lead
-	// to slightly lower fee rates.
-	assertMaxFeeRate(anchorChannel, 1.0, 435941555)
-	assertMaxFeeRate(anchorChannel, 0.001, 435941)
-	assertMaxFeeRate(anchorChannel, 0.000001, 435)
-	assertMaxFeeRate(anchorChannel, 0.0000001, chainfee.FeePerKwFloor)
-}
+	testCases := []struct {
+		name               string
+		channel            *LightningChannel
+		maxFeeAlloc        float64
+		expectedFeeAlloc   float64
+		expectedMinFeeRate bool
+	}{
```

Comment:
I think this is a decent holdover for the time being. The numbers are still magic but restructuring these tests is not the priority for the moment so I think this is a great intermediate result. 🤝